LidarLiteI2C write_reg() correction and uniform initialization.

mcsauder opened the pull request:
**Describe problem solved by the proposed pull request**
This PR corrects a write_reg() usage error in the LidarLite I2C driver that might be causing issues with newer hardware as reported in #12278 and [this forum post](https://discuss.px4.io/t/lidarlite-v3-driver-does-not-automatically-start-with-i2c/10783) (tbd).  Also, an unused `_retries` variable is deleted, and uniform initialization in the base class is employed.

**Test data / coverage**
Bench tests perform identical to master branch.  This PR:
![image](https://user-images.githubusercontent.com/2497951/62334233-11461880-b484-11e9-9a1a-86681db99e5d.png)

PX4:master:
![image](https://user-images.githubusercontent.com/2497951/62334392-b6f98780-b484-11e9-80cc-f9e9b8f5a05f.png)

**Additional context**
See #12278 

Thanks @dakejahl for the potential fix!

mcsauder commented:
@ar1040, could you test this PR and see if it solves your issue?

dakejahl commented on src/drivers/distance_sensor/ll40ls/LidarLite.h:
```diff
@@ -66,35 +66,34 @@ class LidarLite
+	perf_counter_t _sample_interval_perf{perf_alloc(PC_ELAPSED, "ll40ls: interval")};
+	perf_counter_t _sample_perf{perf_alloc(PC_ELAPSED, "ll40ls: read")};
+	perf_counter_t _sensor_resets{perf_alloc(PC_COUNT, "ll40ls: resets")};
+	perf_counter_t _sensor_zero_resets{perf_alloc(PC_COUNT, "ll40ls: zero resets")};
```
`perf_alloc` is dynamic memory allocation, this compiles?

mcsauder replied:
Sure does. Tests fine as well.

dakejahl commented on src/drivers/distance_sensor/ll40ls/LidarLiteI2C.cpp:
```diff
@@ -121,9 +119,6 @@ LidarLiteI2C::probe()
 	uint8_t id_low = 0;
 
-	// more retries for detection
-	_retries = 10;
```
add these back

mcsauder replied:
Done!

mcsauder commented:
Rebased on current master, squashed commits, and bench tested again. Everything looks good from a functional standpoint:
![image](https://user-images.githubusercontent.com/2497951/62337745-25dddd00-b493-11e9-90b9-021c589438c7.png)

mcsauder commented:
Thanks @dagar!

ar1040 commented:
@mcsauder, yes my is working now, but only if I run the _ll40ls start_ command in the Mavlink Console, otherwise it never shows in the analyze window.

HenryHuYu commented:
> **Describe problem solved by the proposed pull request**
> This PR corrects a write_reg() usage error in the LidarLite I2C driver that might be causing issues with newer hardware as reported in #12278 and [this forum post](https://discuss.px4.io/t/lidarlite-v3-driver-does-not-automatically-start-with-i2c/10783) (tbd). Also, an unused `_retries` variable is deleted, and uniform initialization in the base class is employed.
> 
> **Test data / coverage**
> Bench tests perform identical to master branch. This PR:
> ![image](https://user-images.githubusercontent.com/2497951/62334233-11461880-b484-11e9-9a1a-86681db99e5d.png)
> 
> PX4:master:
> ![image](https://user-images.githubusercontent.com/2497951/62334392-b6f98780-b484-11e9-80cc-f9e9b8f5a05f.png)
> 
> **Additional context**
> See #12278
> 
> Thanks @dakejahl for the potential fix!

This phenomenon is not right I think. I can also do this with 2019/5 Garmin lidar lite v3 product. but only in nsh. I use pixhawk and FW 1.9.2 stable. I have set SENS EN LL40LS to I2C. But the driver will never start as I boot the pixhawk. In nsh, It would also never start as  I type' ll40ls start i2c' for the first time, but just like you tested, at secont time or third, it will start.

mcsauder commented:
PR's welcome! :) #12278
